config/load_options.go

```diff
@@ -122,6 +122,10 @@ type LoadOptions struct {
 	// S3UseARNRegion specifies if the S3 service should allow ARNs to direct
 	// the region, the client's requests are sent to.
 	S3UseARNRegion *bool
+
+	// Specifies that SDK clients must resolve a dual-stack endpoint for
+	// services.
+	UseDualStack *bool
```

Comment:
why `*bool` over an Enum?

Reply:
Mainly due to the fact that while we could make this an enum, the clients wouldn't be able to make depend on the declared constants in this package. Alternative would be to declare this an a string enum constant type, and have the resolver interface expose this as just a string value. This would require the logic of interpreting the string values to also be generated into the clients which is one of the downsides.